apps/astarte_realm_management/test/astarte_realm_management/device_removal/core_test.exs

```diff
@@ -0,0 +1,242 @@
+            ) do
+        %IndividualDatastream{
+          device_id: device.device_id,
+          interface_id: interface.interface_id,
+          endpoint_id: endpoint_id,
+          path: "/path/",
+          value_timestamp: value_timestamp,
+          reception_timestamp: reception_timestamp,
+          reception_timestamp_submillis: reception_timestamp_submillis
+        }
+        |> Repo.insert!(prefix: keyspace)
+
+        Core.delete_individual_datastreams!(realm, device.device_id)
+
+        assert Queries.retrieve_individual_datastreams_keys!(realm, device.device_id) == []
+      end
+    end
+
+    property "delete_individual_properties/2 removes individual properties data of a valid device",
+             %{realm: realm} do
+      check all(
+              interface <- Astarte.Core.Generators.Interface.interface(),
+              device <- Astarte.Core.Generators.Device.device(interfaces: interface),
+              endpoint_id = Ecto.UUID.bingenerate()
+            ) do
+        keyspace = Realm.keyspace_name(realm)
+
+        %IndividualProperty{
+          device_id: device.device_id,
+          interface_id: interface.interface_id,
+          endpoint_id: endpoint_id,
+          path: "/example/path"
+        }
+        |> Repo.insert!(prefix: keyspace)
+
+        Core.delete_individual_properties!(realm, device.device_id)
+
+        assert Queries.retrieve_individual_properties_keys!(realm, device.device_id) == []
+      end
+    end
+
+    property "delete_object_datastream/2 removes object datastreams of a valid device", %{
+      realm: realm
+    } do
+      keyspace = Realm.keyspace_name(realm)
+
+      check all(
+              interface <-
+                Astarte.Core.Generators.Interface.interface(
+                  type: :datastream,
+                  aggregation: :object,
+                  explicit_timestamp: true
+                ),
+              device <- Astarte.Core.Generators.Device.device(interfaces: interface),
+              value_timestamp = DateTime.utc_now(),
+              reception_timestamp = DateTime.utc_now(),
+              reception_timestamp_submillis <- integer(0..10)
+            ) do
+        interface_json = Jason.encode!(interface)
+        _ = Engine.install_interface(realm, interface_json)
+
+        table_name =
+          CQLUtils.interface_name_to_table_name(interface.name, interface.major_version)
+
+        """
+        INSERT INTO #{keyspace}.#{table_name}
+        (device_id, path, value_timestamp, reception_timestamp, reception_timestamp_submillis)
+        VALUES (?, ?, ?, ?, ?)
+        """
+        |> Repo.query!([
+          device.device_id,
+          "/example/path",
+          value_timestamp,
+          reception_timestamp,
+          reception_timestamp_submillis
+        ])
+
+        %Device{
+          device_id: device.device_id,
+          introspection: %{interface.name => interface.major_version}
+        }
+        |> Repo.insert!(prefix: keyspace)
```

Comment:
Nit: as agreed upon IRL, consider moving this (and similar) to a `defp` so that the scope of the test is clearer